### ISSUE TITLE
FBSDKSharePhoto.image is nil when sharing an UIImage

### ISSUE THREAD

**Issue by littleylv** [facebook/facebook-sdk-swift#264]
# Facebook Swift SDK Issue

## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-sdk-swift/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Xcode Version: `10.0 beta 5`
- Swift Version: `4.2`
- Installation Platform & Verison: `Cocoapods` version `1.5.3`

### Goals

Share an image.

### Expected Results

Successfully share an image.

### Actual Results

`UserInfo={com.facebook.sdk:FBSDKErrorArgumentValueKey=<FBSDKSharePhoto: 0x280b5f090>, com.facebook.sdk:FBSDKErrorDeveloperMessageKey=Must have an asset, image, or imageURL value., com.facebook.sdk:FBSDKErrorArgumentNameKey=photo}`

### Steps to Reproduce

See code samples.

### Code Samples & Details

In my `ViewController.swift` I have a function:

```swift
let photo = Photo(image: myImage, userGenerated: true)
let content = PhotoShareContent(photos: [photo])
let shareDialog = ShareDialog(content: content)
shareDialog.mode = .native
shareDialog.presentingViewController = self
do {
    try shareDialog.show()
} catch _ as NSError {
}
```

But nothing happens when i click the button, and Xcode console shows the logs: `UserInfo={com.facebook.sdk:FBSDKErrorArgumentValueKey=<FBSDKSharePhoto: 0x280b5f090>, com.facebook.sdk:FBSDKErrorDeveloperMessageKey=Must have an asset, image, or imageURL value., com.facebook.sdk:FBSDKErrorArgumentNameKey=photo}`.

The error comes from [FBSDKSharePhoto.m#L170-L177](https://github.com/facebook/facebook-ios-sdk/blob/master/FBSDKShareKit/FBSDKShareKit/FBSDKSharePhoto.m#L170-L177)

Then I go to [Photo.swift#L67-L70](https://github.com/facebook/facebook-sdk-swift/blob/master/Sources/Share/Content/Photo/Photo.swift#L67-L70), replace them with:

```swift
        print(image)
        photo.image = image
        photo.imageURL = url
        photo.isUserGenerated = isUserGenerated
        photo.caption = caption
        print(photo.image)
        print(photo.imageURL)
        print(photo.isUserGenerated)
        print(photo.caption)
```
Result is:

```
Optional(<UIImage: 0x282e91730>, {1334, 750})
nil
nil
true
nil
```

The weird thing is that `image` is `Optional(<UIImage: 0x282e91730>, {1334, 750})` but after `photo.image = image`, `photo.image` is nil

**Comment by littleylv** [comment]
@lukelucas247 

> The issue doesn't appear and works fine when using previous versions in the profile as below:

Because the `validateWithOptions` method was added in 4.36.0 => [https://github.com/facebook/facebook-ios-sdk/commit/f1c22a5bc22987a3ddad6f51c13fa9f2f4c2a926#diff-db0092fb88ea4b382f14f078489b20fa](https://github.com/facebook/facebook-ios-sdk/commit/f1c22a5bc22987a3ddad6f51c13fa9f2f4c2a926#diff-db0092fb88ea4b382f14f078489b20fa)

**Comment by Beniamiiin** [comment]
It happens because of `imageUrl` setter method. Look at it:
```
internal var sdkPhotoRepresentation: FBSDKSharePhoto {
    let photo = FBSDKSharePhoto()
    photo.image = image
    photo.imageURL = url
    photo.isUserGenerated = isUserGenerated
    photo.caption = caption

    return photo
}
...
- (void)setImageURL:(NSURL *)imageURL
{
    _image = nil;
    _imageURL = [imageURL copy];
    _photoAsset = nil;
}
```

Firstly, we set an image at the line `photo.image = image` and after this, we set an empty `imageUrl` at the line `photo.imageURL = url` and in the setter of this variable `image` set to `nil` if even `imageUrl = nil` too. So, it's a bug.

To fix it, we have to rewrite getter of `sdkPhotoRepresentation ` variable:
```
internal var sdkPhotoRepresentation: FBSDKSharePhoto {
    let photo = FBSDKSharePhoto()
    
    if let image = image {
        photo.image = image
    } else if let imageUrl = url {
        photo.imageURL = imageUrl
    }
    
    photo.isUserGenerated = isUserGenerated
    photo.caption = caption

    return photo
}
```

**Comment by littleylv** [comment]
@Beniamiiin Thanks very much! I have tried it and it works! 👍 

Could you please make a pull request or I can make one if you are busy. 😄

**Comment by codytwinton** [comment]
Resolved in #266
